File: q1-q99/q70-q79/q71/q71.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        path_name = []
        solution = "/"
        i = 0
        size = len(path)
        while i < size:
            word = ""
            while i < size and path[i] != "/":
                word += path[i]
                i += 1
            if word == "..":
                if len(path_name) > 0:
                    path_name.pop()
            elif word != "" and word != ".":
                path_name.append(word)
            i += 1
        for i in path_name:
            solution += i + "/"
        if solution == "/":
            return solution
        return solution[0:len(solution) - 1]

File: Sequence/q1-q99/q70-q79/q71/q71.cpp
class Solution {
public:
    string simplifyPath(string path) {
        int i = 0, size = path.size();
        vector<string> path_name;
        string solution = "/";
        while (i < size) {
            string word = "";
            while (i < size && path[i] != '/') {
                word += path[i++];
            }
            if (word == "..") {
                if (path_name.size() > 0) {
                    path_name.pop_back();
                }
            } else if (word != "" && word != ".") {
                path_name.push_back(word);
            }
            ++i;
        }
        for (auto i : path_name) {
            solution += i + "/";
        }
        return solution == "/" ? solution : solution.substr(0,solution.length()-1);
    }
};
